lib/models/os.coffee

```diff
@@ -304,6 +304,42 @@ getOsModel = (deps, opts) ->
 				throw new Error('No such version for the device type')
 			.asCallback(callback)
 
+	###*
+	# @summary Get an applications config.json
+	# @name getConfig
+	# @public
+	# @function
+	# @memberof resin.models.os
+	#
+	# @param {Number} Id - application id (number)
+	# @fulfil {Object} - application configuration as a JSON object
+	# @returns {Promise}
+	#
+	# @example
+	# resin.models.os.getConfig('MyApp').then(function(config) {
+	# 	fs.writeFile('foo/bar/config.json', JSON.stringify(config));
+	# });
+	#
+	# resin.models.os.getConfig(123).then(function(config) {
+	# 	fs.writeFile('foo/bar/config.json', JSON.stringify(config));
+	# });
+	#
+	#
+	# resin.models.os.download('MyApp', function(error, config) {
+	# 	if (error) throw error;
+	# 	fs.writeFile('foo/bar/config.json', JSON.stringify(config));
+	# });
+	###
+	exports.getConfig = (id, callback) ->
+		callback = findCallback(arguments)
+
+		request.send
+			method: 'GET'
+			url: "/download-config?appId=#{id}"
+			baseUrl: apiUrl
+		.get('body')
+		.asCallback(callback)
```

Comment:
It's nicer to handle 404s by throwing a `ResinApplicationNotFound` error, rather than a ResinRequest error that people will have to introspect, so you can consistency handle that elsewhere. The application model has a bunch of [examples](https://github.com/resin-io/resin-sdk/blob/master/lib/models/application.coffee#L324).